openlibrary/plugins/worksearch/code.py

```diff
@@ -145,7 +145,7 @@ def build_q_list(param):
             q_list.extend(i['op'] if 'op' in i else '%s:(%s)' % (i['field'], i['value']) for i in parse_query_fields(q_param))
         else:
             isbn = normalize_isbn(q_param)
-            if isbn:
+            if isbn and len(isbn) in (10, 13):
```

Comment:
if we wanted even more gains here, we could specifically validate the isbn, but this fixes the majority of the issues.